### ISSUE TITLE
Insert from unnest in .NET core application throws exception

### ISSUE THREAD

**Issue by sergii-van** [crate/crate#10776]
**CrateDB version**: 4.3.1

**Environment description**:  Windows / .NET Core 3.1 / Crate DB 4.3.1 / Npgsql 4.1.5 / Npgsql.EntityFrameworkCore.PostgreSQL 4.1.5

- JVM version: java -version
  ```
  openjdk version "15.0.1" 2020-10-20
  OpenJDK Runtime Environment AdoptOpenJDK (build 15.0.1+9)
  OpenJDK 64-Bit Server VM AdoptOpenJDK (build 15.0.1+9, mixed mode, sharing)
  ```
- Number of nodes: 1
- Using docker: no

**Problem description**:

Running code from steps to reproduce results in the following error:
```
System.NotSupportedException: The CLR array type System.Int32[] isn't supported by Npgsql or your PostgreSQL. If you wish to map it to an  PostgreSQL composite type array you need to register it before usage, please refer to the documentation.
   at Npgsql.TypeMapping.ConnectorTypeMapper.GetByClrType(Type type)
   at Npgsql.NpgsqlParameter`1.ResolveHandler(ConnectorTypeMapper typeMapper)
   at Npgsql.NpgsqlCommand.ValidateParameters(ConnectorTypeMapper typeMapper)
   at Npgsql.NpgsqlCommand.ExecuteReaderAsync(CommandBehavior behavior, Boolean async, CancellationToken cancellationToken)
   at Npgsql.NpgsqlCommand.ExecuteNonQuery(Boolean async, CancellationToken cancellationToken)
```

**The same code on CrateDb 4.2.4 runs without any issues**. Also running query works using Admin UI.

**Steps to reproduce**:

1. Create table // CREATE TABLE mm.data_table (id INTEGER NOT NULL, name TEXT NOT NULL);
2. Create .NET core project
3. Install mentioned packages Npgsql 4.1.5 / Npgsql.EntityFrameworkCore.PostgreSQL 4.1.5
4. Run the following method

```
public async Task TestUnnestAsync()
        {
            var records = Enumerable
                .Range(0, 10)
                .Select(i => (Id: i, Name: $"My identifier is {i}"))
                .ToArray();

            NpgsqlConnection connection = new NpgsqlConnection(connString);            
            connection.Open();

            using var command = new NpgsqlCommand(connection: connection, cmdText: "INSERT INTO mm.data_table (id, name) SELECT * FROM unnest(@i, @n) AS d");

            command.Parameters.Add(new NpgsqlParameter<int[]>("i", records.Select(e => e.Id).ToArray()));
            command.Parameters.Add(new NpgsqlParameter<string[]>("n", records.Select(e => e.Name).ToArray()));

            await command.ExecuteNonQueryAsync();
        }
```

**Comment by amotl** [comment]
Dear @sergii-van,

thanks for reporting this and for sharing such a detailed repro. We will have a look into this.

With kind regards,
Andreas.

**Comment by sergii-van** [comment]
@amotl , does not seems to be related to nullable arrays. Besides, as I mentioned the same code (same environment), but different Crate DB version causes the issue

**Comment by mfussenegger** [comment]
@sergii-van thank you for the detailed report. I can reproduce the regression. We'll come up with a fix for the next 4.3.2 hotfix release.
